ui/components/app/transaction-list/transaction-list.component.js

```diff
@@ -210,15 +267,50 @@ export default function TransactionList({
     useSolanaBridgeTransactionMapping(nonEvmTransactions);
   ///: END:ONLY_INCLUDE_IF
 
-  const unfilteredPendingTransactions = useSelector(
+  const unfilteredPendingTransactionsCurrentChain = useSelector(
     nonceSortedPendingTransactionsSelector,
   );
-  const unfilteredCompletedTransactions = useSelector(
+
+  const unfilteredPendingTransactionsAllChains = useSelector(
+    nonceSortedPendingTransactionsSelectorAllChains,
+  );
+
+  const unfilteredPendingTransactions = useMemo(() => {
```

Comment:
Is it worth decoupling this logic into a hook so we can test it in isolation and reduce the code here?